### DISCUSSION TITLE
Failed to resolve entry for package "monaco-editor"

### DISCUSSION THREAD

**Discussion by zfben** [vitest-dev/vitest#1037] [Q&A]
I build a vite project with the example from https://github.com/microsoft/monaco-editor/tree/main/samples/browser-esm-vite-react .

Then I add vitest and the test code is:

```ts
/**
 * @vitest-environment jsdom
 */

import { render, screen } from '@testing-library/react'
import { Editor } from '..'
import { test, expect } from 'vitest'

test('should work', () => {
  render(< Editor />)

  expect(screen.getByText('Editor')).toEqual({})
})
```

But it shows `Failed to resolve entry for package "monaco-editor". The package may have incorrect main/module/exports specified in its package.json: Failed to resolve entry for package "monaco-editor". The package may have incorrect main/module/exports specified in its package.json`

What should I do to let the case run?

**Comment by sheremet-va** [comment]
This happens because Native Node cannot resolve path to monaco editor because it doesn't have `exports` field. Try adding this to your config:

```ts
{
  test: {
    deps: {
      inline: ['monaco-editor']
    }
  }
}
```

**Reply by sheremet-va** [reply]
I suggest creating an issue in their repository. They don't even have `main` entry in package.json. Their package.json should be something like this:

```json
{
  "main": "./esm/vs/editor/editor.main.js",
  "type": "module",
}
```

**Reply by fneves** [reply]
any way we can mock this dependency from being imported entirely?

**Reply by sheremet-va** [reply]
> any way we can mock this dependency from being imported entirely?

you can use `vi.mock` in `setupFiles`, or `test.alias` in config to redirect to a different file.

**Reply by zfben** [reply]
@fneves Finally, I created an esm package for monaco editor, https://github.com/slashnotes/monaco-editor-esm
It works with vite and doesn't need special configs.
